taichi/program/async_engine.h

```diff
@@ -19,6 +19,62 @@ TLANG_NAMESPACE_BEGIN
 
 // TODO(yuanming-hu): split into multiple files
 
+class IRHandle {
+ public:
+  IRHandle(IRNode const *ir, uint64 hash) : ir_(ir), hash_(hash) {
```

Comment:
nit: I think the majority of the codebase put `const` before type..
```suggestion
  IRHandle(const IRNode *ir, uint64 hash) : ir_(ir), hash_(hash) {
```